### PR TITLE
topics with get route

### PR DESCRIPTION
**Pull request by Alecodeco**
Added Topics model with api get route for authenticated users

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -6,5 +6,8 @@
       registrations: 'api/auth/registrations',
       passwords:     'api/auth/passwords'
     }
+
+    get 'topics', action: :index, controller: 'topics'
```
Comment (enzofab91) [inline config/routes.rb]:
```suggestion
    resources :topics, only: :index
```

File: db/migrate/20210812184207_create_topics.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateTopics < ActiveRecord::Migration[6.1]
+  def change
+    create_table :topics do |t|
+      t.string :name
```
Comment (enzofab91) [inline db/migrate/20210812184207_create_topics.rb]:
```suggestion
      t.string :name, null: false
```

File: app/controllers/api/topics_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module Api  
+  class TopicsController < ActionController::Base
+    include DeviseTokenAuth::Concerns::SetUserByToken
+    before_action :authenticate_api_user!
+
+    def index
+      @topics = Topic.all
```
Comment (enzofab91) [inline app/controllers/api/topics_controller.rb:7]: What about adding a test for this?

**Review by enzofab91 — COMMENTED** [review]
Great work 👏 

Left some comments.

**Review by sebastiancaraballo — COMMENTED** [review]
Please add some tests and also add this [gem](https://github.com/ctran/annotate_models) to annotate models.

File: app/controllers/api/topics_controller.rb (hunk 3 above)
Comment (Alecodeco) [reply app/controllers/api/topics_controller.rb:7]: I'll merge the other tests before that in order to use all the RSpec directories

**Review by enzofab91 — APPROVED** [review]
Good work.

It seems that some commits are not from this feature. Please be careful in next PRs.

**Pull request merged**